src/services/webhook-handler/webhook-queue.service.ts

```diff
@@ -0,0 +1,194 @@
+import { WebhookDeliveryStatus, WebhookEventType } from '@prisma/client';
+import { prisma } from '@/utils/db';
+import { logger } from '@/utils/logger';
+import { webhookSenderService } from './webhook-sender.service';
+
+export class WebhookQueueService {
+  async queueWebhook(
+    eventType: WebhookEventType,
+    payload: Record<string, any>,
+    entityId?: string,
+    paymentSourceId?: string,
+  ): Promise<void> {
+    const webhookEndpoints = await prisma.webhookEndpoint.findMany({
+      where: {
+        isActive: true,
+        disabledAt: null,
+        events: {
+          has: eventType,
+        },
+        ...(paymentSourceId
+          ? {
+              OR: [
+                { paymentSourceId: paymentSourceId },
+                { paymentSourceId: null },
+              ],
+            }
+          : {}),
+      },
+    });
+
+    if (webhookEndpoints.length === 0) {
+      logger.debug('No active webhook endpoints found for event', {
+        event_type: eventType,
+        payment_source_id: paymentSourceId,
+        entity_id: entityId,
+      });
+      return;
+    }
+
+    const webhookPayload = {
+      event_type: eventType,
+      timestamp: new Date().toISOString(),
+      webhook_id: '',
+      data: payload,
+    };
+
+    const deliveries = webhookEndpoints.map(async (endpoint) => {
+      const endpointPayload = {
+        ...webhookPayload,
+        webhook_id: endpoint.id,
+      };
+
+      try {
+        await prisma.webhookDelivery.create({
+          data: {
+            webhookEndpointId: endpoint.id,
+            eventType,
+            payload: endpointPayload as Record<string, any>,
+            entityId,
+            status: WebhookDeliveryStatus.Pending,
+            nextRetryAt: new Date(),
+          },
+        });
+
+        logger.info('Webhook queued for delivery', {
+          webhook_id: endpoint.id,
+          event_type: eventType,
+          entity_id: entityId,
+        });
+      } catch (error) {
+        logger.error('Failed to queue webhook delivery', {
+          webhook_id: endpoint.id,
+          event_type: eventType,
+          entity_id: entityId,
+          error: error instanceof Error ? error.message : 'Unknown error',
+        });
+      }
+    });
+
+    await Promise.allSettled(deliveries);
+  }
+
+  /**
+   * Process pending webhook deliveries
+   */
+  async processPendingDeliveries(): Promise<void> {
+    const pendingDeliveries = await prisma.webhookDelivery.findMany({
+      where: {
+        status: {
```

Comment:
add the is active filter here

Reply:
otherwise it might lead to a bug where it selects 50 inactive ones and does not progress